Snowflake.Data/Core/SFStatement.cs

```diff
@@ -383,11 +383,14 @@ internal async Task<SFBaseResultSet> ExecuteAsync(int timeout, string sql, Dicti
                     SFBindUploader uploader = new SFBindUploader(SfSession, _requestId);
                     await uploader.UploadAsync(bindings, cancellationToken).ConfigureAwait(false);
                     _bindStage = uploader.getStagePath();
-                    ClearQueryRequestId();
                 }
                 catch (Exception e)
                 {
-                    logger.Warn("Exception encountered trying to upload binds to stage. Attaching binds in payload instead. {0}", e);
+                    logger.Warn($"Exception encountered trying to upload binds to stage. Attaching binds in payload instead.", e);

```

Comment:
I think that maybe here we should also explicitly provide the `e.Message,` not the whole stack trace.

Reply:
done